### ISSUE TITLE
[java] ReplaceJavaUtilDate triggers when using class java.sql.Timestamp

### ISSUE THREAD

**Issue by mrclmh** [pmd/pmd#5966]
**Affects PMD Version:** 
7.16.0

**Rule:**
[ReplaceJavaUtilDate](https://docs.pmd-code.org/latest/pmd_rules_java_errorprone.html#replacejavautildate)

**Description:**
The rule triggers when using the `java.sql.Timestamp` class which extends `java.util.Date`.

**Code Sample demonstrating the issue:**

```java
Timestamp ts = template.queryForObject(QUERIES.get("getCurrentTimestamp"),
   EmptySqlParameterSource.INSTANCE, Timestamp.class);
```

**Expected outcome:**
The rule should not trigger.

**Running PMD through:** *[Gradle]*

**Comment by mrclmh** [comment]
Addendum:
There is also `java.sql.Date` and `java.sql.Time`.

I also looked up for some "official" information on deprecation of java.sql.Time/Timestamp/Date and found that [JPA 3.2](https://projects.eclipse.org/projects/ee4j.jpa/releases/3.2) deprecated the old classes:
> Deprecates usage of Calendar, Date, Time, Timestamp, Temporal, MapKeyTemporal and TemporalType in new applications in favour of java.time API

In short:
The rule is correct, the code should be updated.

**Comment by adangel** [comment]
Thanks for verifying - I've created a PR to improve the rule doc (#5967).

**Comment by rturner-edjuster** [comment]
The rationale for closing this doesn't really make sense (refer to https://github.com/pmd/pmd/issues/6025). Please revisit this and reconsider the rationale until such time as there is a new JDBC specification (4.4 is the latest for JDK 25) and the JDBC API includes changes to use the `java.time` classes. Jakarta Persistence is not the JDBC specification which defines `java.sql.Timestamp` (and other related classes). (For context, I would love it if JDBC would migrate, it would avoid our conversion class being needed. However, it's certainly annoying when tooling gives you incorrect information.)
